### PR TITLE
Add interest during construction

### PR DESCRIPTION
**Pull request by matti-lamppu**
# Hitas Pull Request

# Description

- Adds 6% or 14% interest to pre-2011 index calculations on the `/api/v1/housing-companies/{housing_company_id}/apartments/{apartment_id}` endpoint.
- Refactored old SQL strings to django ORM methods. New methods use subqueries instead of aliased joins, and they are much easier to read IMO.
    - In theory, subqueries might be slightly slower due to SQL compilers having better micro-optimization for joins, but I didn't measure any difference with debug-toolbar
- Added conditionals for adding thee interest based on the index calculation wanted.
- Added `select_related` for `financing_method` on the queryset for optimization since `old_hitas_ruleset` is needed to determine if interest should be used.
- Switched from `extra` to `annotations` since extra is [soft-deprecated](https://docs.djangoproject.com/en/4.1/ref/models/querysets/#:~:text=Use%20this%20method%20as%20a%20last%20resort).
- Refactored surface area price ceiling from SQL strings to django ORM methods for consistency with the new index calculations. This also uses subqueries now (see the note above).
- Refactored tests to support `old_hitas_ruleset` as well as pre-2005 construction price index calculations

## Pull request checklist

Check the boxes for each DoD items that has been completed:

- **Frontend**
    - [x] Changes have been tested
- **Backend**
    - [x] Changes have been tested
    - [x] Automatic tests has been added
    - [ ] OpenAPI definitions have been updated
    - [ ] Database migrations will work in test environment
    - [ ] Oracle migration has been updated
    - [ ] initial.json has been updated to work with migrations

## Test plan

- [ ] Checked that unconfirmed maximum prices for apartments that finished construction before 2011/01/01 should now have interest included
    - [ ] For the market price index, it is the 6% interest
    - [ ] For the construction price index, it is the 6% interest if construction finished on or after 2005/01/01, and 14% if before
- [ ] Interest is only calculated if the financing method has the `old_hitas_ruleset` flag. Note that interest is not calculated for pre-2011 apartments if they use new rulesets.

## Tickets

HT-327 - Hinta-arvion laskenta muutokset - rakennusaikainen korko - backend

### CODE REVIEW

File: backend/hitas/views/apartment.py [hunk 1]
```diff
@@ -23,18 +23,18 @@
...
     SurfaceAreaPriceCeiling,
-    MarketPriceIndex,
-    MarketPriceIndex2005Equal100,
-    ConstructionPriceIndex,
-    ConstructionPriceIndex2005Equal100,
 )
 from hitas.models._base import HitasModelDecimalField
 from hitas.models.apartment import ApartmentMarketPriceImprovement, ApartmentState, DepreciationPercentage
-from hitas.utils import this_month, RoundWithPrecision
+from hitas.utils import RoundWithPrecision, this_month
```
Comment (ranta) [inline backend/hitas/views/apartment.py:37]: And this is why the pre-commit hook is amazing (There would be no separate commit for this) 😄

File: backend/hitas/tests/apis/test_api_apartment.py [hunk 2]
```diff
@@ -496,6 +499,10 @@ def _test_max_prices(
         mpi_factory.create(month=now, value=250)
         SurfaceAreaPriceCeilingFactory.create(month=now, value=3000)
 
+    if old_hitas_ruleset:
+        ap.housing_company.financing_method.old_hitas_ruleset = old_hitas_ruleset
```
Comment (ranta) [inline backend/hitas/tests/apis/test_api_apartment.py:510]: This can be defined when creating the apartment like so:
```
ApartmentFactory.create(
    ...,
    building__real_estate__housing_company__financing_method__old_hitas_ruleset = old_hitas_ruleset
)
```

Comment (matti-lamppu) [reply backend/hitas/tests/apis/test_api_apartment.py]: Done

**Review by ranta — APPROVED** [review]
LGTM 🦦

**Comment by matti-lamppu** [comment]
image
feelsgood
